File: 프로그래머스/1/12912. 두 정수 사이의 합/두 정수 사이의 합.cpp
#include <algorithm>

using namespace std;

long solution(int a, int b) {
    long answer = 0;
    int min_val = min(a, b); // Use std::min for minimum value
    int max_val = max(a, b); // Use std::max for maximum value
    
    for (int i = min_val; i <= max_val; i++) {
        answer += i;
    }
    
    return answer;
}

File: 프로그래머스/1/12912. 두 정수 사이의 합/두 정수 사이의 합.java
class Solution {
   public long solution(int a, int b) {
		
		long start = max(a, b);
		long end = min(a, b);
		
        long answer = 0;
        
        for(long i = end;i <= start;i++) {
        	answer += i;
        }
        
        return answer;
    }

	private long max(int a, int b) {
		if(a > b) {
			return a;
		}else {
			return b;
		}
	}
	
	private long min(int a, int b) {
		if(a > b) {
			return b;
		}else {
			return a;
		}
	}
}